doc/nrf/libraries/dfu/suit_dfu.rst

```diff
@@ -0,0 +1,54 @@
+.. _subsys_suit:
+
+Software Updates for Internet of Things (SUIT)
+##############################################
+
+.. contents::
+   :local:
+   :depth: 2
+
+The Software Updates for Internet of Things (SUIT) DFU library provides functionality to the core of a local domain for orchestrating the update based on the SUIT manifest.
+
+Overview
+********
+
+The SUIT DFU library provides APIs to start and prepare the device before the update procedure is started.
+The preparation for the DFU can be described as a sequence inside a SUIT manifest (suit-payload-fetch).
+To interpret and execute this sequence, a SUIT processor must be instantiated inside the firmware of the local domain core.
+This library provides a simplified API that manages the processor state, executes the fetching sequence from the SUIT manifest, and passes the update candidate info to the Secure domain.
+
+The application requires the following steps to perform the Device Firmware Update:
+
+1. Call :c:func:`suit_dfu_initialize` to initialize the SUIT DFU module before any other SUIT functionality is used.
+   This function must be called only once, with at least ``APPLICATION`` init level.
+   Calling this function in ``POST_KERNEL`` or earlier init level causes undefined behavior.
+
+#. If the SUIT manifest requires the fetch of additional data in the suit-payload-fetch sequence,
+   register fetch sources using :c:func:`suit_dfu_fetch_source_register`.
+
+#. Store the SUIT candidate envelope in the non-volatile memory area defined as dfu_partition in the devicetree.
```

Comment:
Is storing in a memory defined as "dfu_partition" a requirement? What would happen if the envelope was stored somewhere else, or the memory area would be called "suit_envelope_partition"?

Reply:
Than this API would not pick it up and trigger the DFU. This library has the partition name hard-coded.

Reply:
Ok, I wasn't expecting a hard-coded partition name. Should we use something like `chosen/nordic,suti-dfu-patition` instead of hardcoding? Of course outside of this PR scope

Reply:
Yes, this is the right way to do that, I'll create a JIRA ticket for it.

Reply:
Please add a link to Jira here for traceability

Reply:
NCSDK-28785